tests/__init__.py

```diff
@@ -551,5 +552,106 @@ def test_batch_lock_key(self):
         Worker(self.tiger).run(once=True)
         self._ensure_queues(queued={'batch': 0})
 
+class TaskTestCase(BaseTestCase):
+    """
+    Task class test cases.
+    """
+    def test_delay(self):
+        task = Task(self.tiger, simple_task)
+        self._ensure_queues()
+        task.delay()
+        self._ensure_queues(queued={'default': 1})
+
+        # Canceling only works for scheduled tasks.
+        self.assertRaises(TaskNotFound, task.cancel)
+
+    def test_delay_scheduled(self):
+        task = Task(self.tiger, simple_task, queue='a')
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'a': 1})
+
+        # Test canceling a scheduled task.
+        task.cancel()
+        self._ensure_queues()
+
+        # Canceling again raises an error
+        self.assertRaises(TaskNotFound, task.cancel)
+
+    def test_delay_scheduled_2(self):
+        task = Task(self.tiger, simple_task, queue='a')
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'a': 1})
+
+        task_id = task.id
+
+        # We can't look up a non-unique task by recreating it.
+        task = Task(self.tiger, simple_task, queue='a')
+        self.assertRaises(TaskNotFound, task.cancel)
```

Comment:
The exception name is a bit misleading in this case - `TaskNotFound` doesn't actually mean that the task wasn't found here - it means that it can't be identified w/o providing its ID. Maybe it would be better to raise `TaskLookupError` or something like that in this case?